presto-hive/src/main/java/com/facebook/presto/hive/HiveCoercer.java

```diff
@@ -128,8 +132,9 @@ public Block apply(Block block)
         }
 
         @Override
-        public TupleDomainFilter toCoercingFilter(TupleDomainFilter filter)
+        public TupleDomainFilter toCoercingFilter(TupleDomainFilter filter, Subfield subfield)
         {
+            checkArgument(subfield.getPath().isEmpty(), "Subfields on primitive types are not allowed");
```

Comment:
Do we wanna have a helper function called `isPrimitive()` in `Subfield` given it will be used multiple times?

Reply:
@highker Something like this would be useful, but I can't think of a good name just yet. `isPrimitive` doesn't work because `a IS NULL` might represent column `a` of any type, primitive or not.